app/src/main/java/com/kodeco/android/countryinfo/ui/components/CountryInfoList.kt

```diff
@@ -1,27 +1,79 @@
 package com.kodeco.android.countryinfo.ui.components
 
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentHeight
 import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
-import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.kodeco.android.countryinfo.flow.Flows
 import com.kodeco.android.countryinfo.model.Country
-import com.kodeco.android.countryinfo.util.CountryInfoState
 
-// TODO fill out CountryInfoList
 @Composable
-fun CountryInfoList(countryList: List<Country>, updateDetails: (CountryInfoState) -> Unit) {
+fun CountryInfoList(countryList: List<Country>, onRefreshClick: () -> Unit) {
+
+    var selectedCountry: Country? by rememberSaveable { mutableStateOf(null) }
+
+    Column {
+        // TODO: Implement the Row composable here that contains the
+        //  the tap/back flow data and the refresh button.
+
+        Row(modifier = Modifier
```

Comment:
Minor but you could move this Row to it's own composable/file.  That way the logic for this composable, including the references to the Flow objects is encapsulated in it's own composable function rather than being a direct part of CountryInfoList.